### ISSUE TITLE
Events V2 filters: Custom date range fields render red error messaging / outlines on initial load

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#9418]
## Describe the defect
Slack thread: https://dsva.slack.com/archives/C52CL1PKQ/p1655135457289439?thread_ts=1654637841.727989&cid=C52CL1PKQ

As a user, I don't expect to see red validation / error messages until after I have entered invalid or incomplete data, and submitted a form. 

On Events V2 filters, red field validation text is shown when you first open the filter, e.g., “Please provide a valid date”. 
A better user experience would be to display only after the user enters something invalid and submits.

## To Reproduce
1. Go to https://www.va.gov/outreach-and-events/events/
2. Click on Filter By: Custom date range
3. See error

## Expected behavior
Form opens with date fields empty, no error message or red outlines. 

## Screenshots
![image (1)](https://user-images.githubusercontent.com/85581471/173399290-0dd928bc-6277-459b-b6e1-d299ff7f3bd4.png)

## Desktop (please complete the following information if relevant, or delete)
All browsers, versions.

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by maxx1128** [comment]
I took a few pre-emptive attempts to fix this and didn't have much luck.

1. I made the date fields not required until the first search was done so they couldn't validate as wrong. But that made the words "required" not appear next to the dates until the search is run, which would also confuse users.
2. Prepopulating the date fields would solve this as well, but is likely a larger UX discussion instead of just a bug fix.
3. The date components can accept functions for `onDateChange` and `onDateBlur`, but stubbing those out didn't solve the issue. My guess is the validation functions are handled separately but not in any available parameters (https://design.va.gov/storybook/?path=/docs/components-va-date--default)
